From 452c2ae77f6977664f7390a71bdc73f1732e18f6 Mon Sep 17 00:00:00 2001
From: Sonny Sasaka <sonnysasaka@chromium.org>
Date: Tue, 27 Mar 2018 22:38:46 -0700
Subject: [PATCH] dbus: Make MockObjectManager useful

MockObjectManager is currently not useful as the mocked methods are not
virtual.

BUG=chromium:812468
TEST=None
---
 dbus/object_manager.h | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/dbus/object_manager.h b/dbus/object_manager.h
index a7bd5cb79635..035d5693d3b7 100644
--- a/dbus/object_manager.h
+++ b/dbus/object_manager.h
@@ -192,35 +192,35 @@ public:
   // interface named in |interface_name|. That object's CreateProperties()
   // method will be used to create instances of dbus::PropertySet* when
   // required.
-  void RegisterInterface(const std::string& interface_name,
-                         Interface* interface);
+  virtual void RegisterInterface(const std::string& interface_name,
+                                 Interface* interface);
 
   // Unregister the implementation class for the D-Bus interface named in
   // |interface_name|, objects and properties of this interface will be
   // ignored.
-  void UnregisterInterface(const std::string& interface_name);
+  virtual void UnregisterInterface(const std::string& interface_name);
 
   // Returns a list of object paths, in an undefined order, of objects known
   // to this manager.
-  std::vector<ObjectPath> GetObjects();
+  virtual std::vector<ObjectPath> GetObjects();
 
   // Returns the list of object paths, in an undefined order, of objects
   // implementing the interface named in |interface_name| known to this manager.
-  std::vector<ObjectPath> GetObjectsWithInterface(
+  virtual std::vector<ObjectPath> GetObjectsWithInterface(
       const std::string& interface_name);
 
   // Returns a ObjectProxy pointer for the given |object_path|. Unlike
   // the equivalent method on Bus this will return NULL if the object
   // manager has not been informed of that object's existance.
-  ObjectProxy* GetObjectProxy(const ObjectPath& object_path);
+  virtual ObjectProxy* GetObjectProxy(const ObjectPath& object_path);
 
   // Returns a PropertySet* pointer for the given |object_path| and
   // |interface_name|, or NULL if the object manager has not been informed of
   // that object's existance or the interface's properties. The caller should
   // cast the returned pointer to the appropriate type, e.g.:
   //   static_cast<Properties*>(GetProperties(object_path, my_interface));
-  PropertySet* GetProperties(const ObjectPath& object_path,
-                             const std::string& interface_name);
+  virtual PropertySet* GetProperties(const ObjectPath& object_path,
+                                     const std::string& interface_name);
 
   // Instructs the object manager to refresh its list of managed objects;
   // automatically called by the D-Bus thread manager, there should never be
-- 
2.17.0.rc1.321.gba9d0f2565-goog

